src/Navbar.js

```diff
@@ -57,7 +58,36 @@ const propTypes = {
    * @controllable navExpanded
    */
   onToggle: React.PropTypes.func,
-
+  /**
+   * A callback fired when a `<NavItem>` grandchild is selected inside a child
```

Comment:
What about "...when a descendant of a child `<Nav>` is selected..."? This might refer to NavItem, LinkContainer, etc., basically any descendants that can be *selected* without referring to any specific ones.